splice: bad commitment signature

From CI
[80.zip](https://github.com/ElementsProject/lightning/files/13068705/80.zip)

Seems related to https://github.com/ElementsProject/lightning/issues/6810#event-10752401849 as these appear to both have to do with post-splice channel confirmation timing.

Relevant log, occurring immediately after state changing to `CHANNELD_AWAITING_SPLICE` to `CHANNELD_NORMAL`:

> FAILED tests/test_splicing.py::test_splice_out - pyln.client.lightning.RpcError: RPC call failed: method: pay, payload: {'bolt11': 'lnbcrt1n1pjnv93nsp5ruhvcutw7lras80x8mhetsa0zzlw7rk6qwdtyr6sty3xnhqnztnqpp5am76ft7tsvla88myg6t443367qguegyzwjxn9jn598al0y2y2dtsdq8deh47vcxqyjw5qcqp99qxpqysgqt7j7hfzu8awk7k87vlyndzg7kxvyf4hzl0juldnt3csgf6wkjk0qqzdh8svxp9nvd8wla5v08uf4zz0hcjrymv60qzyq3ww4ef6hx6qqwn9k4m'}, error: {'code': 210, 'message': '**Destination 022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59 is not reachable directly and all routehints were unusable.**', 'attempts': [{'status': 'failed', 'failreason': 'Destination 022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59 is not reachable directly and all routehints were unusable.', 'partid': 0, 'amount_msat': 100msat}]}

--- a/tests/test_splicing.py
+++ b/tests/test_splicing.py
@@ -35,6 +35,7 @@ def test_splice(node_factory, bitcoind):
 
     l2.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
     l1.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+    l1.daemon.wait_for_log(r'private channel announcement from channeld for ' + first_scid(l1, l2))
 
     inv = l2.rpc.invoice(10**2, '3', 'no_3')
     l1.rpc.pay(inv['bolt11'])
@@ -129,6 +130,7 @@ def test_splice_listnodes(node_factory, bitcoind):
 
     l2.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
     l1.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+    l1.daemon.wait_for_log(r'private channel announcement from channeld for ' + first_scid(l1, l2))
 
     bitcoind.generate_block(7)
 
@@ -162,6 +164,7 @@ def test_splice_out(node_factory, bitcoind):
 
     l2.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
     l1.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+    l1.daemon.wait_for_log(r'private channel announcement from channeld for ' + first_scid(l1, l2))
 
     inv = l2.rpc.invoice(10**2, '3', 'no_3')
     l1.rpc.pay(inv['bolt11'])
@@ -217,6 +220,7 @@ def test_invalid_splice(node_factory, bitcoind):
 
     l2.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
     l1.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+    l1.daemon.wait_for_log(r'private channel announcement from channeld for ' + first_scid(l1, l2))
 
     inv = l2.rpc.invoice(10**2, '3', 'no_3')
     l1.rpc.pay(inv['bolt11'])
@@ -265,6 +269,7 @@ def test_commit_crash_splice(node_factory, bitcoind):
 
     l2.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
     l1.daemon.wait_for_log(r'CHANNELD_AWAITING_SPLICE to CHANNELD_NORMAL')
+    l1.daemon.wait_for_log(r'private channel announcement from channeld for ' + first_scid(l1, l2))
 
     time.sleep(1)
 

